### PR TITLE
LPS-122873

### PR DESCRIPTION
**Pull request by RitaSilveira**
JIRA Ticket:
LPS-122873
LPS-136539

Motivation:
The relative order assigned to the categories represented in the chart must remain throughout the user's session so there is no confusion

### CODE REVIEW

**Comment by RitaSilveira** [comment]
ci:test:sf

**Comment by RitaSilveira** [comment]
ci:test:sf

**Comment by RitaSilveira** [comment]
ci:test:tango

**Comment by RitaSilveira** [comment]
ci:test:sf

**Comment by RitaSilveira** [comment]
ci:test:tango

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 1]
```diff
@@ -1782,6 +2056,69 @@ definition {
 		}
 	}
 
+	@description = "LPS-122873 When the user flip the audit graph axes, the relative ordering of the categories remain the same"
+	@priority = "5"
+	test ReversingTheAxesUsingTheFlipButton {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2115]: This test is not needed, there is already a test testing this, called FlipAxesButton. You can delete it.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 2]
```diff
@@ -113,6 +258,67 @@ definition {
 		}
 	}
 
+	@description = "LPS-122873 When all the web content is unassigned from a Vocab1's category, the relative ordering of the rest of the categories remain the same"
+	@priority = "5"
+	test Category2OfVocab1NotSelected {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:264]: Change the title of this test, is not easy to know what it does. Dont refer it to concrete names, it has to say something related with the order or something like that

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 3]
```diff
@@ -73,6 +157,67 @@ definition {
 		}
 	}
 
+	@description = "LPS-122873 When there are some categories created for vocabulary1 (X axis), they will always have the same order during the user session"
+	@priority = "5"
+	test CategoriesForVocab1 {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:162]: Change the title of this test, is not easy to know what it does. Dont refer it to concrete names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 4]
```diff
@@ -38,6 +38,90 @@ definition {
 		}
 	}
 
+	@description = "LPS-122873 When a new category is created and assigned to a web content, the relative ordering of categories remain the same and the new category gets a new color, positionated in the last position"
+	@priority = "5"
+	test ANewCategory {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:43]: Change the title of this test, is not easy to know what it does.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 5]
```diff
@@ -2033,6 +2370,150 @@ definition {
 		}
 	}
 
+	@description = "LPS-122873 When any web content is unassigned to a Vocab1's category, the relative ordering of categories remain the same"
+	@priority = "5"
+	test SlelectSomeCategoriesFromVocab1 {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2429]: Change the title of this test, is not easy to know what it does. Dont refer it to concrete names. Also it has a mistake in the Select

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 6]
```diff
@@ -2063,4 +2544,60 @@ definition {
 			title = "WC Title");
 	}
 
+
+	@description = "LPS-122873 When the user add filters need to have only one vocabulary, the relative ordering of the categories remain the same"
+	@priority = "5"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Also review the priority of all of the tests. All of them are Critical=FP5??
Remember:
* Critical: FP5
* High: FP4
* Medium: FP3

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by RitaSilveira** [comment]
fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 7]
```diff
@@ -38,6 +38,90 @@ definition {
...
+				vocabularyName = "Stage");
+
+			PortletEntry.publish();
+		}
+		task ("Navigate to the content panel and see if, in the new category order, the new category receives a new color, placed in the last position") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Here you say that it navigates and check the category order, color and last position, but you are just navigating. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 8]
```diff
@@ -73,6 +157,67 @@ definition {
...
+
+				PortletEntry.publish();
+			}
+		}
+		task ("Navigate to content Dashboard and View the order of categories") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:174]: Here you say that it navigates and check the category order but you are just navigating. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase (hunk 2 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:180]: This name remains the same, please modify.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 9]
```diff
@@ -113,6 +258,67 @@ definition {
...
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:235]: Here you say that it navigates and check the category order but you are just navigating. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 10]
```diff
@@ -1552,6 +1758,61 @@ definition {
...
+
+				PortletEntry.publish();
+			}
+		}
+		task ("Go to content Dashboard and View the relative ordering of the categories remain the same") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1892]: Here you say that it navigates and check the category order but you are just navigating. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 11]
```diff
@@ -2033,6 +2361,150 @@ definition {
...
+				}
+
+				for (var count : list "1,2,3") {
+					WebContent.selectCategory(
+						categoryName = "test ${count}",
+						vocabularyName = "Stage");
+				}
+
+				PortletEntry.publish();
+			}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Can this be done in a macro?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 12]
```diff
@@ -2033,6 +2361,150 @@ definition {
...
+		}
+		task ("Navigate to content Dashboard and View the order of categories") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Here you say that it navigates and check the category order but you are just navigating. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 13]
```diff
@@ -2033,6 +2361,150 @@ definition {
...
+		task ("Navigate to content Dashboard, uncheck a vocabulary category 2 and see the order of categories") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+
+			Click.javaScriptClick(
+				checkboxName = "test 2",
+				locator1 = "Checkbox#ANY_CHECKBOX");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2505]: Here you say that it navigates and check the category order but you are just navigating and unchecking the checkbox. Where are the asserts?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 14]
```diff
@@ -2033,6 +2361,150 @@ definition {
...
+				}
+
+				for (var count : list "1,2,3") {
+					WebContent.selectCategory(
+						categoryName = "test ${count}",
+						vocabularyName = "Stage");
+				}
+
+				PortletEntry.publish();
+			}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2494]: Can this be done in a macro?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 15]
```diff
@@ -1602,6 +1863,73 @@ definition {
...
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1997]: Can this be done in a macro?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 16]
```diff
@@ -113,6 +258,67 @@ definition {
...
+				}
+
+				for (var count : list "1,2,3") {
+					WebContent.selectCategory(
+						categoryName = "test ${count}",
+						vocabularyName = "Stage");
+				}
+
+				PortletEntry.publish();
+			}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:228]: Can this be done in a macro?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 17]
```diff
@@ -73,6 +157,67 @@ definition {
...
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:167]: Can this be done in a macro?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 18]
```diff
@@ -38,6 +38,90 @@ definition {
...
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Can this be done in a macro?

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by CarlosBrichete** [comment]
hey @RitaSilveira, please review my comments

**Comment by RitaSilveira** [comment]
fixed

**Comment by CarlosBrichete** [comment]
ci:test:tango

**Comment by CarlosBrichete** [comment]
ci:test:tango

**Comment by CarlosBrichete** [comment]
ci:forward

**Pull request closed without merging**